packages/cosmic-swingset/src/block-manager.js

```diff
@@ -75,7 +76,12 @@ export default function makeBlockManager({
       default:
         assert.fail(X`${action.type} not recognized`);
     }
-    p.then(finish, finish);
+    p.then(finish, e => {
+      // None of these must fail, and if they do, log them verbosely before
+      // returning to the chain.
+      console.error(action.type, 'error:', e);
+      finish();
```

Comment:
this looks like it's discarding any error that might appear in a rejection.. does the caller know enough to abort if one of these functions fails?

Reply:
That's what it _looks like_, but I can tell you that it isn't.  I've added comments.

Using `p.then(...)` attaches callbacks but that doesn't affect the original promise (`p`).  Then I `return p`, the original promise (which still either resolves or rejects as it did before).